apps/common-app/src/new_api/complicated/camera/index.tsx

```diff
@@ -0,0 +1,287 @@
+import React, { useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import {
+  useLongPressGesture,
+  usePinchGesture,
+  useTapGesture,
+  useCompetingGestures,
+  GestureDetector,
+  useSimultaneousGestures,
+  usePanGesture,
+  useExclusiveGestures,
+} from 'react-native-gesture-handler';
+import Animated, {
+  SharedValue,
+  runOnJS,
+  useAnimatedProps,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import { Circle, Svg } from 'react-native-svg';
+import AnimatedCameraView from '../../../common_assets/AnimatedCameraView/AnimatedCameraView';
+import { COLORS } from '../../../common';
+
+const FILTERS = [
+  COLORS.RED,
+  COLORS.GREEN,
+  COLORS.NAVY,
+  COLORS.YELLOW,
+  COLORS.KINDA_BLUE,
+];
+const CAROUSEL_SIZE = 100;
+const FILTER_SIZE = 60;
+const VIDEO_DURATION = 20000;
+const RECORD_INDICATOR_STROKE = 10;
+
+const AnimatedCircle = Animated.createAnimatedComponent(Circle);
+
+export default function Camera() {
+  const [facing, setFacing] = useState<'front' | 'back'>('back');
+  const selectedFilter = useSharedValue(0);
+  const captureProgress = useSharedValue(0);
+  const zoom = useSharedValue(1);
+
+  const filterChangeGesture = usePanGesture({
+    onUpdate: (e) => {
+      'worklet';
+      selectedFilter.value -= e.changeX / FILTER_SIZE;
+    },
+    onDeactivate: () => {
+      'worklet';
+      const nextFilter = Math.min(
+        FILTERS.length - 1,
+        Math.max(0, Math.round(selectedFilter.value))
+      );
+      selectedFilter.value = withTiming(nextFilter, { duration: 150 });
+    },
+  });
+
+  function takePhoto() {
+    alert("I didn't bother to implement this :)");
+  }
+
+  function startRecording() {
+    // no-op
+  }
+
+  function stopRecording() {
+    alert("I didn't bother to implement this either :)");
+  }
+
+  const takePhotoGesture = useTapGesture({
+    onDeactivate: () => {
+      'worklet';
+      runOnJS(takePhoto)();
+      captureProgress.value = withTiming(0, { duration: 1000 });
+    },
+  });
+
+  const takeVideoGesture = useLongPressGesture({
+    shouldCancelWhenOutside: false,
+    maxDistance: 10000,
+    onActivate: () => {
+      'worklet';
+      runOnJS(startRecording)();
+      captureProgress.value = withTiming(1, { duration: VIDEO_DURATION });
```

Comment:
I think this is missing `easing: Easing.linear` - the animation speeds up and slows down.